src/sensor_fusion.c

```diff
@@ -0,0 +1,346 @@
+/**
+ * \brief Function Definition Library to implement Sensor Fusion Algorithm
+ * \author jaspalsingh@cmail.carleton.ca
+ * \version 1.0
+ * \date 2019-12-01
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+#include <gsl/gsl_math.h>
+#include <gsl/gsl_eigen.h>
+#include "sensor_fusion.h"
+
+/**
+ * \brief Structure pointer for computed support degree matrix
+ * \details Structure pointer for calculating support degree matrix
+ * \param[in] sensor_readings array of sensor readings
+ * \return Success: pointer to the structure support_degree_matrix
+ * \return Failure: NULL
+ */
+struct support_degree_matrix *compute_support_degree_matrix(float *sensor_readings) {
+    struct support_degree_matrix *spd;
+    spd = (struct support_degree_matrix *)malloc(sizeof(struct support_degree_matrix));
+    // count the number of sensor readings
+    int length = sizeof(sensor_readings) / sizeof(sensor_readings[0]);
+    spd->sensor_count = length;
+    double **arrptr = (double **)malloc(length * sizeof(double *));
```

Comment:
Be consistent with your variable names and use snake case (arrptr should be arr_ptr)

Reply:
Fixed in recent commits to dev branch